pkg/api/server/v1alpha2/results_test.go

```diff
@@ -147,6 +149,112 @@ func TestGetResult(t *testing.T) {
+		expect    *pb.Result
+		errcode   codes.Code
+	}{
+		{
+			name:   "Test no Mask",
+			in:     &pb.Result{Name: "foo/results/bar-001"},
+			update: &pb.Result{Annotations: map[string]string{"foo": "bar"}},
+			expect: &pb.Result{Annotations: map[string]string{"foo": "bar"}},
+		},
+		{
+			name:      "Test Mask with empty field",
+			in:        &pb.Result{Name: "foo/results/bar-002"},
+			fieldmask: &field_mask.FieldMask{Paths: []string{}},
+			// unset field value to default value in fieldmask
+			update: &pb.Result{Name: "foo/results/bar-002", Annotations: map[string]string{"foo": "bar"}},
+			expect: &pb.Result{},
+		},
+		{
+			name:      "Test Mask with nil Paths field",
+			in:        &pb.Result{Name: "foo/results/bar-003"},
+			fieldmask: &field_mask.FieldMask{},
+			// do not update
+			update: &pb.Result{Name: "foo/results/bar-003", Annotations: map[string]string{"foo": "bar"}},
+			expect: &pb.Result{},
+		},
+
+		// Errors
+		{
+			name:      "ERR Test update with invalid name",
+			in:        &pb.Result{Name: "foo/results/bar-005"},
+			fieldmask: &field_mask.FieldMask{Paths: []string{"annotations"}},
+			// do not update
+			update:  &pb.Result{Name: "invalid", Annotations: map[string]string{"foo": "bar"}},
+			errcode: codes.NotFound,
+		},
+		{
+			name:      "ERR Test Mask with invalid mask field",
+			in:        &pb.Result{Name: "foo/results/bar-004"},
+			fieldmask: &field_mask.FieldMask{Paths: []string{"annotations", "invalid_field"}},
+			// do not update
+			update:  &pb.Result{Name: "foo/results/bar-004", Annotations: map[string]string{"foo": "bar"}},
+			errcode: codes.NotFound,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			parent, _, err := result.ParseName(tc.in.GetName())
+			if err != nil {
+				t.Fatalf("could not parse result name: %v", err)
+			}
+			r, err := srv.CreateResult(ctx, &pb.CreateResultRequest{Result: tc.in, Parent: parent})
+			if err != nil {
+				t.Fatalf("could not create taskrun: %v", err)
+			}
+
+			// If we're doing a full update, pass through immutable fields to
+			// the update result. Since these are created dynamicly,
+			// we can't prepopulate these.
+			if tc.fieldmask == nil {
+				tc.update.Id = r.GetId()
+				tc.update.Name = r.GetName()
+				tc.update.CreatedTime = r.GetCreatedTime()
+			}
+			// Update the created taskrun
+			r, err = srv.UpdateResult(ctx, &pb.UpdateResultRequest{Result: tc.update, Name: tc.update.GetName(), UpdateMask: tc.fieldmask})
+			if err != nil {
+				if status.Code(err) != tc.errcode {
+					t.Fatalf("expected error: %v, got %v", tc.errcode, status.Code(err))
+				} else {
+					return
+				}
+				t.Fatalf("could not update taskrun: %v, %v", err, status.Code(err))
+			}
+
+			// Expected results should always match the created result.
+			if tc.expect != nil {
+				tc.expect.Name = tc.in.GetName()
+				tc.expect.Id = r.GetId()
+				tc.expect.CreatedTime = r.CreatedTime
+				tc.expect.UpdatedTime = r.UpdatedTime
```

Comment:
Let's set an explicit time here - if we didn't actually update the time, this would still pass as is. Don't forget to increment the fake clock!